src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -294,7 +301,9 @@ public ExponentialHistogramData GetExponentialHistogramData()
             this.ThrowNotSupportedMetricTypeException(nameof(this.GetExponentialHistogramData));
         }
 
-        return this.mpComponents!.Base2ExponentialBucketHistogram.GetExponentialHistogramData();
+        Debug.Assert(this.mpComponents?.Base2ExponentialBucketHistogram != null, "Base2ExponentialBucketHistogram was null");
```

Comment:
Use `this.mpComponents!` in `Debug.Assert`?

Reply:
My goal here is to assert in debug all the stuff we just assume to be non-null and access with suppression using `!`.

So I wouldn't do this...

```csharp
Debug.Assert(this.mpComponents!.Base2ExponentialBucketHistogram != null, "Base2ExponentialBucketHistogram was null");
```

Because if someone made a mistake upstream and `this.mpComponents` somehow ended up `null` it could yield a NRE.

But we could do this...

```csharp
        Debug.Assert(this.mpComponents != null, "this.mpComponents was null");
        Debug.Assert(this.mpComponents!.HistogramBuckets != null, "this.mpComponents.HistogramBuckets was null");
```

The reason I went with the `?` in there is it more or less does the work of the two assertions in one shot.